cpp/include/raft/distance/detail/chebyshev.cuh

```diff
@@ -72,7 +72,7 @@ static void chebyshevImpl(const DataT* x,
 
   // Accumulation operation lambda
   auto core_lambda = [] __device__(AccT & acc, DataT & x, DataT & y) {
-    const auto diff = raft::L1Op<AccT, IdxT>()(x - y);
+    const auto diff = raft::myAbs(x - y);
```

Comment:
It sould be nice to see this become a `raft::abs_op` and the one below it become `raft::max_op`. Any reason for now consoliding these as well?

Reply:
Should we not replace all these calls with the standard equivalents, e.g `myAbs` becomes `std::abs` which works on both host and device? The current definition of `myAbs` will compile with nvcc but not gcc.

Reply:
My thinking was just to wrap the functions w/ the `raft::` namespace (even if just calling down to the std::) to keep the APIs consistent and provide a facade to centralized future changes as needed. So long as there's no perf impact for doing so.

Reply:
I see. One objection that I have about putting them in the same header as the functors, though, is that they achieve different purposes: the functors are meant to be callables used as lambdas / passed to prims, not to be called directly. `myXxx` (->`raft::xxx`) is used directly as a device/type-agnostic wrapper around some math functions. It could be confusing to mix them.

Reply:
That sounds fine to me. What about a header like `raft/core/math.hpp`? Again, we don't have to do it in this PR, but it's nice seeing all of these APIs consolidated and put in more obvious places that are easier to document.

Reply:
Would it make sense to replace all cases with `std::xxx` alternatives where possible, as @Nyrio suggested, and only add new functions to raft when there are no suitable standard alternatives? I'm afraid otherwise a user may be confused as to why they should prefer raft version over std for some basic primitives. Or, as a compromise, we can just re-export some of these primitives (e.g. `namespace raft { using std::max; }`).

Reply:
I'm definitely not against re-exporting into the raft namespace but my suggestion is mostly conditional on whether there's ever going to be a perf hit from using std vs some custom accelerated intrinsic or function that invokes CUDA-specific instructions. In that case, we could still export the functions into the RAFT namespace but explicltly define new raft variants when necessary. Overall, I'm trying to provide as consistent an experience as possible in the raft namespace to prevent users from having to refer to the docs to figure out which things are covered under RAFT and which things are not. not saying we need to export everything in std math, but the things which we frequently use would be useful to keep in the `raft::` namespace even if only as a facade to enable us to centralize changes in the future.

Reply:
> whether there's ever going to be a perf hit from using std vs some custom accelerated intrinsic

With nvcc, writing `std::abs` uses the `std` version on the host side, and the device intrinsics on the device side. So no perf hit compared to the current code. (with Clang I've seen compilation errors because it says we're trying to use host functions in device code). Artem's suggestion is simply an easier way to expose these names as part of the `raft` namespace without unnecessary wrappers, but that still gives up the option to write different variants later.

Reply:
It's probably better to make this change in a separate PR.